### DISCUSSION TITLE
How do I add a remote?

### DISCUSSION THREAD

**Discussion by llimllib** [Vinzent03/obsidian-git#628] [Q&A]
I'm attempting to follow the [getting started doc](https://publish.obsidian.md/git-doc/Getting+Started)'s "Start with new repo" section to back up my obisidan repo to an existing, empty, github repository. Here's what I did:

1. Install the plugin
2. run the "initialize a new repo" command
3. run "edit remotes"

and here's where I get stuck. The text on that command says "select or create a new remote by typing its name and selecting it", but I cannot figure out how to create a remote; I've tried putting the URL in and hitting enter, putting the URL in and selecting it, putting a remote name instead of a URL and doing the same things, and nothing seems to work.

No matter what I do, I get "error: no such remote".

I'm a frequent github user, and have OS X authentication set up properly.

Do I need to go into the repository and add the remote manually, or is there a way to do this within obsidian that I'm not seeing?

**Comment by llimllib** [comment]
OK I followed the code a bit:

- [edit remotes](https://github.com/denolehov/obsidian-git/blob/cc4cc5ebcbf5668d4584fa2cad70f8638b10704f/src/main.ts#L1590) calls `getRemoteUrl`
- [getRemoteUrl](https://github.com/denolehov/obsidian-git/blob/master/src/gitManager/simpleGit.ts#L524) runs `git remote get-url <name>` against the remote name you passed in
- this throws an uncaught exception if the remote name does not exist

So I don't understand how that command could possibly create a remote - it fails as soon as it finds that the remote name you've entered does not exist?

**Reply by llimllib** [reply]
I see that the code exists [a few lines later](https://github.com/denolehov/obsidian-git/blob/cc4cc5ebcbf5668d4584fa2cad70f8638b10704f/src/main.ts#L1606) to add a remote, but this code never gets hit. This is a bug, right?

**Reply by llimllib** [reply]
To be more precise, `getRemoteUrl` seems to be returning a promise that rejects, not throwing an exception, but the result is the same - the code ceases execution and never proceeds to allow you to add a remote

**Comment by Vinzent03** [comment]
There was an issue with creating a remote, which has been fixed now. You can add a remote by calling the edit remotes command and typing the name of a not existing remote.
